src/api/Types.ts

```diff
@@ -6,5 +6,21 @@
 import { AppMetadata } from '..';
 import { Context } from '../context/ContextTypes';
 
+/**
+ * Operations that target apps (like open or raiseIntent) can identify
+ * an app just by by its name, or pass full app metadata, giving the
+ * desktop agent more information about the targeted app.
+ */
 export type TargetApp = string | AppMetadata;
+/**
+ * Describes a callback that handles a context event.
+ * Used when attaching listeners for context broadcasts.
+ */
 export type ContextHandler = (context: Context) => void;
+/**
+ * Describes a callback that handles a context event and may return a
+ * promise of a Context object to be returned to the application that
+ * raised the intent.
```

Comment:
I guess we could also add that if void is returned by the IntentHandler a Promise will still be still returned to the intent caller - but it will only mean that the intent was properly dispatched and its execution started.

Reply:
That was the intent (pun intended ;-) ). I'll attempt to clarify.

```suggestion
 * raised the intent. Any response other than Promise indicates that the handler does not return data (which will result in the Desktop Agent rejecting the IntentResolution.getData() promise with the appropriate error message).
```

Reply:
Does that help @bertrand-s ?